src/NzbDrone.Core/Update/InstallUpdateService.cs

```diff
@@ -153,7 +186,29 @@ private void EnsureAppDataSafety()
             if (_appFolderInfo.StartUpFolder.IsParentPath(_appFolderInfo.AppDataFolder) ||
                 _appFolderInfo.StartUpFolder.PathEquals(_appFolderInfo.AppDataFolder))
             {
-                throw new NotSupportedException("Update will cause AppData to be deleted, correct you configuration before proceeding");
+                throw new UpdateFailedException("Update will cause AppData to be deleted, correct you configuration before proceeding");
+            }
+        }
+
+        private void ExecuteInstallUpdate(Command message, UpdatePackage package)
+        {
+            try
+            {
+                InstallUpdate(package);
+
+                message.Completed("Restarting Sonarr to apply updates");
+            }
+            catch (UpdateFolderNotWritableException ex)
+            {
+                message.Failed(ex, string.Format("Startup folder not writable by user '{0}'", Environment.UserName));
+            }
+            catch (UpdateVerificationFailedException ex)
+            {
+                message.Failed(ex, "Update verification failed");
```

Comment:
Why can't it be too long, messenger doesn't seem to truncate it.

![image](https://cloud.githubusercontent.com/assets/412280/5906215/c0358d4c-a54a-11e4-8267-1396a7a4b9f8.png)